.github/workflows/build.yml

```diff
@@ -308,3 +310,194 @@ jobs:
           commit-message: publish documentation
           clean-exclude: |
             "version/*"
+
+  upload-nightly:
+    if: (github.ref == 'refs/heads/master') && (github.event_name == 'schedule')
+    needs: [test-unix, test-macos, test-windows]
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/download-artifact@v2
+      with:
+        name: wheels
+        path: dist/
+
+    - run: |
+        zip -r dist.zip dist/
+
+    - uses: actions/github-script@v5
+      id: asset
+      with:
+        github-token: ${{ secrets.gh_access_token }}
+        script: |
+          const fs = require('fs');
+
+          // Get the ref for HEAD [A call to the API is probably useless]
+          let ref_head = await github.rest.git.getRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: 'heads/master',
+          });
```

Comment:
As mentioned in comment, there is no need for an API call, this could be written:
```js
    const master_sha = '${{ github.sha }}';
```
I renamed ref_head to master_sha because HEAD refers to the tip of any branch, not master.